test/functorch/test_vmap.py

```diff
@@ -3286,6 +3286,9 @@ def test():
     @toleranceOverride({torch.float32: tol(atol=1e-04, rtol=1e-04)})
     @skipOps('TestVmapOperatorsOpInfo', 'test_vmap_exhaustive', vmap_fail.union({
         xfail('cat'),
+        xfail('nn.functional.conv1d'),  # AssertionError: Exception not raised on ErrorInput
```

Comment:
cc @zou3519

Reply:
Sure, @zou3519! I'd love to work on fixing this bug. Will update that in this PR. Thanks!

Reply:
Concretely there is a "separate bias path" where we just add the output of convolution with the bias where there is no error checking, so I think an error check in https://github.com/pytorch/pytorch/blob/4b757f4633494d7bbc55973f36f14aeca96387fa/aten/src/ATen/functorch/BatchRulesConvolution.cpp#L108 matching however the existing error check looks like in the conv code would be sufficient

Reply:
I see. Thanks for your input, @zou3519! Adding there.

Reply:
@zou3519, I think there are some other problems here too. As in I implemented keeping an equal number of output channels and bias shape, I see we are still getting error on the functorch side. Please see:
```python
In [1]: import torch

In [2]: from torch import vmap

In [3]: input = torch.rand((1, 1, 4, 4))

In [4]: weight = torch.rand((1, 1, 3, 2))

In [5]: bias = torch.rand((1,))

In [6]: vmap(torch.nn.functional.conv2d)(input, weight, bias)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [6], in <cell line: 1>()
----> 1 vmap(torch.nn.functional.conv2d)(input, weight, bias)

File ~/Documents/ldev/pytorch/torch/_functorch/vmap.py:434, in vmap.<locals>.wrapped(*args, **kwargs)
    430     return _chunked_vmap(func, flat_in_dims, chunks_flat_args,
    431                          args_spec, out_dims, randomness, **kwargs)
    433 # If chunk_size is not specified.
--> 434 return _flat_vmap(
    435     func, batch_size, flat_in_dims, flat_args, args_spec, out_dims, randomness, **kwargs
    436 )

File ~/Documents/ldev/pytorch/torch/_functorch/vmap.py:39, in doesnt_support_saved_tensors_hooks.<locals>.fn(*args, **kwargs)
     36 @functools.wraps(f)
     37 def fn(*args, **kwargs):
     38     with torch.autograd.graph.disable_saved_tensors_hooks(message):
---> 39         return f(*args, **kwargs)

File ~/Documents/ldev/pytorch/torch/_functorch/vmap.py:619, in _flat_vmap(func, batch_size, flat_in_dims, flat_args, args_spec, out_dims, randomness, **kwargs)
    617 try:
    618     batched_inputs = _create_batched_inputs(flat_in_dims, flat_args, vmap_level, args_spec)
--> 619     batched_outputs = func(*batched_inputs, **kwargs)
    620     return _unwrap_batched(batched_outputs, out_dims, vmap_level, batch_size, func)
    621 finally:

RuntimeError: expected stride to be a single integer value or a list of 1 values to match the convolution dimensions, but got stride=[1, 1]

In [7]: torch.nn.functional.conv2d(input, weight, bias)
Out[7]: 
tensor([[[[3.6119, 2.7547, 2.2168],
          [2.7564, 2.9878, 2.1065]]]])
```
I raised an issue earlier here: #85656.

Can we please land this PR, and I'll send a follow-up patch for this bug?

I think @zou3519 is on leave. @Chillee, could you please take a look at this? Thank you very much!